### ISSUE TITLE
[no-action-on-submit-button] Do not handle `<form method="dialog">`

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#2989]
For forms with `method="dialog"` the submit button *must* have a click-handler as it doesn't do anything otherwise.

cc @JoaoDsv, @bertdeblock, @bmish

**Comment by boris-petrov** [comment]
The [Examples](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-action-on-submit-button.md#examples) for `forbids` and `allows` should be swapped when the form has a `method="dialog"` attribute. Because in that case the form isn't expected to submit a request but rather have some JavaScript handler.

**Comment by boris-petrov** [comment]
No, not exactly. The idea is that `method="dialog"` means that there is no backend-processing for that form. So - without a submit-handler on the form or click-handler on the submit button - it would do nothing. The `no-action-on-submit-button` rule requires submit buttons to have no handlers but in the `method="dialog"` case the *opposite* is true - a handler **is** mandatory.

**Comment by lifeart** [comment]
To clarify, if form have `method="dialog"`, we should emit error in following cases:

1. Submit action is added to form (we can't submit dialogs)
2. Button type='submit' added inside this form
3. Button type='button' don't have 'on click' handler

**Comment by angelayanpan** [comment]
this issue can be closed? Looks like PR merged
